network/p2p/inspector/validation/control_message_validation_inspector.go

```diff
@@ -209,6 +209,23 @@ func (c *ControlMsgValidationInspector) Inspect(from peer.ID, rpc *pubsub.RPC) e
 		return nil
 	}
 
+	// check peer identity when running private network
+	// sanity check: rpc inspection should be disabled on public networks
+	if c.networkingType == network.PrivateNetwork {
+		err := c.checkSenderIdentity(from)
+		if err != nil {
+			c.notificationConsumer.OnInvalidControlMessageNotification(p2p.NewInvalidControlMessageNotification(from, p2pmsg.CtrlMsgGraft, err, 1, p2p.CtrlMsgNonClusterTopicType))
```

Comment:
It seems we need to add a `p2pmsg.Rpc` as well for the notification type. Also, as a future work, we may rename this `InvalidControlMessageNotification` to `RpcInspectionFailureNotification` as it is turning to a more generic structure covering the `rpc`, `publish` and other control messages.

Reply:
https://github.com/onflow/flow-go/pull/5449/commits/b8d83cc3f586a72da632d9959f2c35b4fa9ea3b4